src/new-contract-interaction/extract_addresses.js

```diff
@@ -0,0 +1,26 @@
+// extract all the token addresses from aave's mainnet.json file
+
+const fs = require('fs');
+
+const tokenAddressesData = fs.readFileSync('./mainnet.json');
```

Comment:
Should this be read from a file?  Or should this be pulled directly from the Aave ProtocolDataProvider contract at runtime?

The second question definitely involves more complexity, so I understand the simplicity of sticking with a static file.
```
const protocolDataProviderContract = new ethers.Contract(
  protocolDataProviderAddress, protocolDataProviderAbi, jsonRpcProvider,
);
let reserveTokenArray = await protocolDataProviderContract.getAllReservesTokens();
```

Reply:
This is a one-time configuration step to pull addresses out of mainnet.json; extract_addresses.js is run manually when mainnet.json is updated.  Doing it dynamically at runtime is a nice idea but I would defer that to a future enhancement if we find that the tokens are frequently changing.

Reply:
I made the handler read the aToken addresses dynamically using the protocol data provider contract.  It turns out that the mainnet.json file is out of sync with what is running on the blockchain, so this turned out to be a valuable change.